modules/miner/blockmanager.go

```diff
@@ -69,21 +69,53 @@ func (m *Miner) HeaderForWork() (types.BlockHeader, types.Target) {
 	lockID := m.mu.Lock()
 	defer m.mu.Unlock(lockID)
 
-	// Grab a block for work.
-	block, target := m.blockForWork()
+	var header types.BlockHeader
+	var randTxn types.Transaction
+	var block *types.Block
 
-	// Save a mapping between the block and its header, replacing the block
```

Comment:
typically (within this project - idk about other projects), the variables are put directly above the codeblock that first uses them.

```
// comment explaining vars + codeblock
var x y
var z a
if {
} else {
}
```